File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int m = grid.size();
        vector<vector<int>>ans(m-2,vector<int>(m-2,0));
        for(int i=1;i<m-1;i++){
            for(int j=1;j<m-1;j++){
                int p = INT_MIN;
                for(int k = i-1;k<=i+1;k++){
                    for(int m = j-1;m<=j+1;m++){
                        p = max(p,grid[k][m]);
                    }
                }
                ans[i-1][j-1]=p;
            }
        }
        return ans;
    }
};

File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.py
class Solution(object):
    def largestLocal(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: List[List[int]]
        """
        n = len(grid)
        ans = []
        for i in range(1,n-1):
            tempr = []
            for j in range(1,n-1):
                temp = 0
                for k in range(i-1,i+2):
                    for m in range(j-1, j+2):
                        temp = max(temp,grid[k][m])
                tempr.append(temp)

            ans.append(tempr)

        return ans


        